aws/resource_aws_instance.go

```diff
@@ -1146,9 +1146,9 @@ func readBlockDevicesFromInstance(instance *ec2.Instance, conn *ec2.EC2) (map[st
 		}
 		if vol.VolumeType != nil {
 			bd["volume_type"] = *vol.VolumeType
-		}
-		if vol.Iops != nil {
-			bd["iops"] = *vol.Iops
+			if "io1" == strings.ToLower(*vol.VolumeType) && vol.Iops != nil {
```

Comment:
Also, I would prefer to write the state in both `io1` & `gp2` cases: when `gp2` is used, the iops are 3 times the volume size with a minimum of 100gb. So if you have a 34GB volume, it would result in 104 iops.
I think it's worth keeping that information for people wanting to know it.

Thoughts?

Reply:
@Ninir  Regarding your second comment, a problem I see with keeping state for `gp2` volume iops is that anybody managing EBS volumes with Terraform would have to keep track of the iops multiplier and update 2 values when adding/changing `gp2` volumes.
IMHO, there's little discernible benefit to having to manage this number unless it is `io1`.

It might also create confusion for users who think they could change iops on a `gp2` volume when they cannot.

Reply:
Updated to use the type and perform the calculation for gp2.

Reply:
Thanks for that @fraenkel!

@mrwacky42 Could you clarify the part about `anybody managing EBS volumes with Terraform would have to keep track of the iops multiplier and update 2 values when adding/changing gp2 volumes` please?
Your point seems interesting, however not sure to get it right! 😅 

Thanks!

Reply:
@Ninir - 
The current expectation is that I do not have to specify iops for `gp2` volumes because it is meaningless. I worry that mapping values for `gp2` volume iops would force me to change my Terraform config from:

```
resource "aws_instance" "foo" {
  ...
  root_block_device {
    volume_type = "gp2"
    volume_size = "500"
  }
}
```

to include iops in order to avoid Terraform thinking the resource is changed on every run:
```
resource "aws_instance" "foo" {
  ...
  root_block_device {
    volume_type = "gp2"
    volume_size = "500"
    iops = "1500"
  }
}
```

However, I could be misunderstanding the how the code change affects everything, so feel free to disregard!

See also the PR and issue mentioned in the comments:
https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_instance.go#L1388-L1394